trends_presentation/src/commonMain/kotlin/net/thechance/mena/trends/presentation/screen/show_real/component/ReelCard.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+import coil3.compose.AsyncImage
+import mena.trends_presentation.generated.resources.Res
+import mena.trends_presentation.generated.resources.ic_eyee
+import mena.trends_presentation.generated.resources.ic_dots
+import mena.trends_presentation.generated.resources.ic_heart
+import mena.trends_presentation.generated.resources.just_now
+import mena.trends_presentation.generated.resources.likes
+import mena.trends_presentation.generated.resources.likes_suffix
+import mena.trends_presentation.generated.resources.more_options
+import mena.trends_presentation.generated.resources.profile_image
+import mena.trends_presentation.generated.resources.video_thumbnail
+import mena.trends_presentation.generated.resources.views
+import mena.trends_presentation.generated.resources.views_suffix
+import net.thechance.mena.designsystem.presentation.component.icon.Icon
+import net.thechance.mena.designsystem.presentation.component.text.Text
+import net.thechance.mena.designsystem.presentation.theme.theme.Theme
+import net.thechance.mena.trends.presentation.screen.show_real.TrendsScreenState
+import net.thechance.mena.trends.presentation.screen.show_real.extention.toTimeAgo
+import net.thechance.mena.trends.presentation.shared.util.isValidImageUrl
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.resources.stringResource
+
+@Composable
+internal fun TrendCard(
+    trend: TrendsScreenState.TrendUiState,
+    onMoreClick: () -> Unit,
+    onLikeClick: () -> Unit,
+    onVideoClick: (String) -> Unit
+) {
+    val timeAgoText = trend.timeAgo?.toTimeAgo() ?: stringResource(Res.string.just_now)
+
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clip(RoundedCornerShape(Theme.radius.lg))
+            .background(Theme.colorScheme.background.surfaceLow)
+            .padding(bottom = Theme.spacing._12)
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = Theme.spacing._12, vertical = Theme.spacing._8),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            AsyncImage(
+                model = if (isValidImageUrl(trend.profileImageUrl)) trend.profileImageUrl else null,
+                contentDescription = stringResource(Res.string.profile_image),
+                modifier = Modifier
+                    .size(40.dp)
+                    .clip(CircleShape),
+                contentScale = ContentScale.Crop
+            )
+
```

Comment:
can we did this for more readablity?
```suggestion
 ProfileImage(profileImageUrl = trend.profileImageUrl)
 
 //in the end of the file 
 @Composable
private fun ProfileImage(profileImageUrl: String,modifier: Modifier= Modifier){
    AsyncImage(
        model = if (isValidImageUrl(profileImageUrl)) profileImageUrl else null,
        contentDescription = stringResource(Res.string.profile_image),
        modifier = modifier
            .size(40.dp)
            .clip(CircleShape),
        contentScale = ContentScale.Crop
    )
}